src/form/time/input/TimeInput.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {useState} from "react";
+import classNames from "classnames";
+
+import Input from "../../input/Input";
+import {DATE_FORMAT} from "../../../core/utils/time/timeConstants";
+import {formatDateWithOptions, parseTime} from "../../../core/utils/time/timeUtils";
+
+export interface TimeInputProps {
+  testid: string;
+  onChange: (timeString: string) => void;
+  selectedDate?: Date | null;
+  isDisabled?: boolean;
+  placeholder?: string;
+  name?: string;
+  icon?: React.ReactNode;
+  hasError?: boolean;
+  customClassName?: string;
+}
+
+function TimeInput({
+  testid,
+  selectedDate,
+  onChange,
+  isDisabled = false,
+  placeholder = "03:30 PM",
+  name = "",
+  icon,
+  customClassName,
+  hasError
+}: TimeInputProps) {
+  const timeStringOfDate = selectedDate
+    ? formatDateWithOptions({
+        format: DATE_FORMAT.LONG_TIME_FORMAT,
+        shouldShiftDateToCompensateForTimezone: false
+      })(selectedDate)
+    : "";
+  const [value, setValue] = useState(timeStringOfDate);
```

Comment:
I think it should also get a `value` prop to turn this component into a controlled component. When the value is passed from the parent, TimeInput should delegate the handling of its state to its parent. Also, it seems we want to calculate `timeStringOfDate` just once.
```suggestion
  const [value, setValue] = useState(() => selectedDate
    ? formatDateWithOptions({
        format: DATE_FORMAT.LONG_TIME_FORMAT,
        shouldShiftDateToCompensateForTimezone: false
      })(selectedDate)
    : "");
```